ring/token_generator.go

```diff
@@ -35,16 +43,17 @@ func (t *RandomTokenGenerator) GenerateTokens(requestedTokensCount int, allTaken
 		return []uint32{}
 	}
 
-	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-
 	used := make(map[uint32]bool, len(allTakenTokens))
 	for _, v := range allTakenTokens {
 		used[v] = true
 	}
 
 	tokens := make([]uint32, 0, requestedTokensCount)
 	for i := 0; i < requestedTokensCount; {
-		candidate := r.Uint32()
+		t.m.Lock()
```

Comment:
Why do we need this?

Reply:
Because access to `*math.Rand` from different goroutines needs to be serialized.

Reply:
I see, previously we used to create a new instance of `Rand` inside the method itself, and it was used by only 1 go routine. But it is not the case anymore.